components/agenda/src/Agenda.svelte

```diff
@@ -161,26 +169,48 @@
       themeUrl = _this.theme;
     }
 
+    if (!isValidTimezone(_this.timezone)) {
+      console.warn(`Invalid IANA time zone: ${_this.timezone}`);
+      _this.timezone = undefined;
+    } else if (
+      _this.timezone === Intl.DateTimeFormat().resolvedOptions().timeZone
+    ) {
+      _this.timezone = undefined;
+    }
+
     if (_this.selected_date) {
       const date = convertToUTC(new Date(_this.selected_date));
       date.setHours(0, 0, 0, 0);
       selectedDate = date;
     } else if (allowedDates?.length) {
       selectedDate = allowedDates[0];
     } else {
-      const date = new Date();
+      let date = new Date();
+      if (_this.timezone) {
+        date = new Date(getTimeInTimezone(date, _this.timezone));
+      }
```

Comment:
If you convert this to unix, you don't need to set the value of date again
```suggestion
      let date = new Date().getTime();
```

Reply:
How do I setHours if I convert it to unix?